internal/envtest/environment.go

```diff
@@ -75,6 +77,56 @@ func init() {
 	utilruntime.Must(admissionv1.AddToScheme(scheme.Scheme))
 }
 
+// RunTestsInput is the input for RunTests.
+type RunTestsInput struct {
+	M                   *testing.M
+	Env                 *Environment
+	ManagerUncachedObjs []client.Object
+	SetupIndexes        func(ctx context.Context, mgr ctrl.Manager)
+	SetupReconcilers    func(ctx context.Context, mgr ctrl.Manager)
+}
+
+// RunTests executes the tests of the given testing.M in a test environment.
+// Note: The environment will be created in this func and should not be created before. This func takes a *Environment
+//       because our tests require access to the Environment. We use this field to make the created Environment available
+//       at the call site.
+// Note: Test environment creation can be skipped by setting the environment variable `CAPI_DISABLE_TEST_ENV`. This only
+//       makes sense when executing tests which don't require the test environment, e.g. tests using only the fake client.
+func RunTests(ctx context.Context, input RunTestsInput) int {
+	if os.Getenv("CAPI_DISABLE_TEST_ENV") != "" {
+		return input.M.Run()
+	}
```

Comment:
Is this still necessary if we go down the route of supporting a pre-existing kind environment?

Reply:
I personally would prefer it as its nice to just start the already configured test run/debug configuration in the IDE without any additional setup at all.